test/e2e/operator_test.go

```diff
@@ -3579,6 +3579,36 @@ func waitForDeploymentEnvVar(t *testing.T, cl client.Client, deployment *appsv1.
 	return err
 }
 
+// waitForDeploymentCompleteWithCleanup waits for a deployment to complete its rollout, then waits for the old
+// generation's pods to finish terminating.
+func waitForDeploymentCompleteWithCleanup(t *testing.T, cl client.Client, deploymentName types.NamespacedName, timeout time.Duration) error {
+	t.Helper()
+	deployment := &appsv1.Deployment{}
+	if err := cl.Get(context.TODO(), deploymentName, deployment); err != nil {
+		return fmt.Errorf("failed to get deployment %s: %w", deploymentName.Name, err)
+	}
+
+	if deployment.Generation != deployment.Status.ObservedGeneration {
+		if err := waitForDeploymentComplete(t, cl, deployment, timeout); err != nil {
+			return fmt.Errorf("timed out waiting for deployment %s to complete rollout: %w", deploymentName.Name, err)
+		}
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		return fmt.Errorf("deployment %s has invalid spec.selector: %w", deploymentName.Name, err)
+	}
+
+	return wait.PollImmediate(2*time.Second, timeout, func() (bool, error) {
+		pods := &corev1.PodList{}
```

Comment:
s/pods/pod.

Reply:
`pods` is a list of possibly multiple pods, so I think plural makes more sense. That said, if it's unclear as is, I do think `podList` as a name would work.

Reply:
In my head I read this as a `podList`, but `Items` brings the plurality.